--- /emsdk/upstream/emscripten/system/lib/libc/musl/src/time/__tz.c	2024-07-22 03:41:44.000000000 +0000
+++ /root/__tz.c	2024-08-01 18:13:42.962355411 +0000
@@ -114,11 +114,6 @@

 #define VEC(...) ((const unsigned char[]){__VA_ARGS__})

-static uint32_t zi_read32(const unsigned char *z)
-{
-	return (unsigned)z[0]<<24 | z[1]<<16 | z[2]<<8 | z[3];
-}
-
 static size_t zi_dotprod(const unsigned char *z, const unsigned char *v, size_t n)
 {
 	size_t y;
@@ -131,6 +126,11 @@
 }
 #endif

+static uint32_t zi_read32(const unsigned char *z)
+{
+	return (unsigned)z[0]<<24 | z[1]<<16 | z[2]<<8 | z[3];
+}
+
 static void do_tzset()
 {
 #ifdef __EMSCRIPTEN__
@@ -280,7 +280,6 @@
 #endif
 }

-#ifndef __EMSCRIPTEN__
 /* Search zoneinfo rules to find the one that applies to the given time,
  * and determine alternate opposite-DST-status rule that may be needed. */

@@ -440,7 +439,6 @@
 	*zonename = __tzname[1];
 	UNLOCK(lock);
 }
-#endif

 static void __tzset()
 {
